### ISSUE TITLE
Testing and further updates to Kronos

### ISSUE THREAD

**Issue by mikemcka** [mahmoodlab/KRONOS#10]
Hi,

Thankyou for making KRONOS publicly available. My Group has recently acquired a Lunaphore COMET system and we are very keen to give KRONOS a try for phenotyping and all the other functions it can perform.

I have a few questions however.

1. For the CHL codex dataset used in the repo as example data for training (https://zenodo.org/records/10067010), is the a corresponding nuclear segmentation mask available that could be shared alongside the membrane segmentation. I wish to do a comparison with our current process where we use QuPath features combined with some custom generated features for cell phenotyping using ML. Currently I cannot generate the same feature set as we usually do without the nuclear segmentation. I can generate some nuclear segmentation myself using deepcell from the instructions from the supplementary information from the paper https://www.nature.com/articles/s41467-023-44188-w#data-availability  but before I go down that rabbit hole I was wondering if you had them available to share? Currently I only have internally completed MIBI projects to test but KRONOS is designed for fluorescence data so that will likely not work well. I have also been looking online and on zenodo and researchers are often only providing the raw images but either not the segmentation or cell type annotation and vice versa. Any help would be great.

2.  I have been doing some experiments with the machine learning step of the cell phenotyping tutorial with using some different machine learning methods with the extracted feature embeddings from the model, and I have found that in comparison with the code from the tutorial I could quite quickly outperform the Linear regression model results generated with the provided notebooks. I used an XGBoost model with a very small hyperparameter search with some class balancing using SMOTE resampling and saw increases in F1 scores of 5% for most of the cell type class metrics. Did you choose linear regression as your ML model because you found it was performing the best or did you just find use it as an example and expect users to perform their own ML optimisation. All good either way, just wanted to know what you think and what you find works well.

3. With the recent release of Dino v3 https://ai.meta.com/blog/dinov3-self-supervised-vision-model/ did you have any plans to update the current dino v2 model provided with a v3 model with the same dataset you used to train previously? 

Thankyou again for providing this very promising and exciting tool. 

Many thanks,

Mike

**Comment by andrewsong90** [comment]
Dear @mikemcka 

Thank you for your interest in KRONOS and sincere apologies for such a delay in getting back to you.

1. We have used MESMER for cell segmentation. By now, you would probably have come to resolution anyways, but integrating MESMER and running it should be straightforward! In the future releases, we will look into what is possible to release for these public datasets. 

2. You are right in pointing out that XGBoost (or MLPs) can/should outperform the linear classifier baseline. The goal of KRONOS (as with general Foundation models in vision and pathology) is to provide a very informative embedding of the corresponding image patch as an input to any downstream models or architectures (which are user-defined). The choice of linear probe is to demonstrate the "raw quality" of such embeddings, as this is the simplest model that can be used on top of KRONOS embeddings. This also allows fair comparison between different baselines as well. If MLP or XGBoost was used on top, it would be harder to delineate whether the good performance is coming from the KRONOS embedding itself or the additional XGBoost/MLP.

3. Yes hopefully in the future! We are working on updating KRONOS.

Hope this helps after very long delay!
Andrew

**Comment by mikemcka** [comment]
Hi Andrew,

Thanks for getting back to me. Great to know I am on the right track. Thankyou for the clear explanation of the intentions of the tool. I will have some labelled data shortly so very keen to explore what the embeddings can do!

I have noticed that the model generates quite a lot of features will larger panels, did you ever experiment with any dimensionality reduction methods to reduce your dataset size prior to training? I will explore this anyway when I get started so all good if you didn't.

Great you will continue to update the model as well. Really appreciate it.

Many thanks,

Mike

**Comment by andrewsong90** [comment]
Hi @mikemcka 

Thank you for supporting KRONOS.

Yes, we are making updates to the model - Along with the better pretraining strategies, one of the updates is to make sure that our KRONOS features are of fixed dimension and thus agnostic to the input number of markers, to prevent confusion & overfitting.

So stay tuned!

Best,
Andrew
